### ISSUE TITLE
atsign-doc can't be used with macros

### ISSUE THREAD

**Issue by timholy** [JuliaLang/julia#9259]
```julia
julia> @doc "A function" f() = 3
f (generic function with 1 method)

julia> @doc f
  A function

julia> @doc "An inlined function" @inline g() = 4
(anonymous function)

julia> g()
ERROR: g not defined

julia> @doc g
ERROR: g not defined

julia> @inline h() = 6
h (generic function with 1 method)

julia> h()
6
```

**Comment by fcard** [comment]
I know how to fix this for macros with one or more arguments, but how do you propose the results of zero-argument macros be documented? Currently `@doc "something" @m` documents `@m` itself, not the result of `@m()`.

(@mschauer's solution doesn't work anymore, because this inconsistency between the two notations has been fixed.)

**Comment by fcard** [comment]
This has been fixed by #12000.

**Comment by yuyichao** [comment]
Right.
